applications/doodle/src/doodle_inbound_listener_sup.erl

```diff
@@ -0,0 +1,109 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2013, 2600Hz
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%-------------------------------------------------------------------
+-module(doodle_inbound_listener_sup).
+
+-behaviour(supervisor).
+
+-export([start_link/0]).
+-export([init/1]).
+
+-include("doodle.hrl").
+
+-define(JSON(Proplist), {Proplist}).
+
+-define(DEFAULT_EXCHANGE, <<"sms">>).
+-define(DEFAULT_EXCHANGE_TYPE, <<"topic">>).
+-define(DEFAULT_EXCHANGE_OPTIONS, [{'passive', 'true'}] ).
+-define(DEFAULT_EXCHANGE_OPTIONS_JOBJ, ?JSON(?DEFAULT_EXCHANGE_OPTIONS) ).
+
+-define(DEFAULT_BROKER, wh_amqp_connections:primary_broker()).
+-define(QUEUE_NAME, <<"smsc_inbound_queue_", (?DOODLE_INBOUND_EXCHANGE)/binary>>).
+
+-define(DOODLE_INBOUND_QUEUE, whapps_config:get_ne_binary(?CONFIG_CAT, <<"inbound_queue_name">>, ?QUEUE_NAME)).
+-define(DOODLE_INBOUND_BROKER, whapps_config:get_ne_binary(?CONFIG_CAT, <<"inbound_broker">>, ?DEFAULT_BROKER)).
+-define(DOODLE_INBOUND_EXCHANGE, whapps_config:get_ne_binary(?CONFIG_CAT, <<"inbound_exchange">>, ?DEFAULT_EXCHANGE)).
+-define(DOODLE_INBOUND_EXCHANGE_TYPE, whapps_config:get_ne_binary(?CONFIG_CAT, <<"inbound_exchange_type">>, ?DEFAULT_EXCHANGE_TYPE)).
+-define(DOODLE_INBOUND_EXCHANGE_OPTIONS,  whapps_config:get(?CONFIG_CAT, <<"inbound_exchange_options">>, ?DEFAULT_EXCHANGE_OPTIONS)).
+
+
+-define(ATOM(A), wh_util:to_atom(A, 'true')).
+-define(CHILD_NAME, ?ATOM(wh_util:rand_hex_binary(16))).
+
+-define(CHILDREN, [?WORKER_NAME_ARGS('doodle_inbound_listener', ?CHILD_NAME, [C]) 
+                     || #amqp_listener_connection{}=C <- connections()
+                  ]).
+
+%% ===================================================================
+%% API functions
+%% ===================================================================
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Starts the supervisor
+%% @end
+%%--------------------------------------------------------------------
+-spec start_link() -> startlink_ret().
+start_link() ->
+    supervisor:start_link({'local', ?MODULE}, ?MODULE, []).
+
+%% ===================================================================
+%% Supervisor callbacks
+%% ===================================================================
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Whenever a supervisor is started using supervisor:start_link/[2,3],
+%% this function is called by the new process to find out about
+%% restart strategy, maximum restart frequency and child
+%% specifications.
+%% @end
+%%--------------------------------------------------------------------
+-spec init([]) -> sup_init_ret().
+init([]) ->
+    wh_util:set_startup(),
+    RestartStrategy = 'one_for_one',
```

Comment:
true but i wanted to supply the CHILDREN list and that's not possible in simple_one_for_one. maybe there's another strategy for initializing the CHILDREN list ?